notif.go

```diff
@@ -34,7 +34,7 @@ func (nn *netNotifiee) Connected(n network.Network, v network.Conn) {
 		if dht.host.Network().Connectedness(p) == network.Connected {
 			bootstrap := dht.routingTable.Size() <= minRTBootstrapThreshold
 			dht.Update(dht.Context(), p)
-			if bootstrap && dht.triggerAutoBootstrap {
+			if bootstrap && dht.autoBootstrap {
```

Comment:
So the first 4 peers that we add to the routing table will all trigger a bootstrap each back to back. And `triggerBootstrap` is not a buffered channel, so "Connected" notifications 2-4 may block if the bootstrap triggered by 1 is still in progress? Maybe I'm misreading, but this seems like a concurrency bug.

Reply:
Forget it, I glossed over the default branch 🙃